src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -133,12 +155,74 @@ func (a *metricsAppender) SamplesAppender(opts SampleAppenderOptions) (SamplesAp
 			})
 		}
 
-		stagedMetadatas := matchResult.ForExistingIDAt(nowNanos)
-		if !stagedMetadatas.IsDefault() && len(stagedMetadatas) != 0 {
-			a.debugLogMatch("downsampler applying matched mapping rule",
+		// Always aggregate any default staged metadatas (unless
+		// mapping rule has provided an override for a storage policy,
+		// if so then skip aggregating for that storage policy).
+		for _, stagedMetadatas := range a.defaultStagedMetadatas {
+			a.debugLogMatch("downsampler applying default mapping rule",
 				debugLogMatchOptions{Meta: stagedMetadatas})
 
-			// Only sample if going to actually aggregate
+			stagedMetadatsBeforeFilter := stagedMetadatas[:]
+			if len(a.mappingRuleStoragePolicies) != 0 {
+				// If mapping rules have applied aggregations for
+				// storage policies then de-dupe so we don't have two
+				// active aggregations for the same storage policy.
+				stagedMetadatasAfterFilter := stagedMetadatas[:0]
+				for _, stagedMetadata := range stagedMetadatas {
+					pipesAfterFilter := stagedMetadata.Pipelines[:0]
+					for _, pipe := range stagedMetadata.Pipelines {
+						storagePoliciesAfterFilter := pipe.StoragePolicies[:0]
+						for _, sp := range pipe.StoragePolicies {
+							// Check aggregation for storage policy not already
+							// set by a mapping rule.
+							matchedByMappingRule := false
+							for _, existing := range a.mappingRuleStoragePolicies {
+								if sp.Equivalent(existing) {
+									matchedByMappingRule = true
+									a.debugLogMatch("downsampler skipping default mapping rule storage policy",
```

Comment:
Should this also log exactly which rule was skipped?

Reply:
Might get confusing since it could be a subsection of it - thankfully we print that out just as we start matching it (so its contextually close in the logs)
```
			a.debugLogMatch("downsampler applying default mapping rule",
				debugLogMatchOptions{Meta: stagedMetadatas})
```

Reply:
Fair